### PR TITLE
optimize add by ~30%

### PR DESCRIPTION
**Pull request by debris**
old: 

```
test u256_add      ... bench:     135,355 ns/iter (+/- 30,832)
test u512_add      ... bench:     213,261 ns/iter (+/- 44,027)
```

new:

```
test u256_add      ... bench:     107,183 ns/iter (+/- 14,998)
test u512_add      ... bench:     148,255 ns/iter (+/- 25,584)
```

### CODE REVIEW

File: src/uint.rs [hunk 1]
```diff
@@ -63,21 +63,30 @@ macro_rules! uint_overflowing_add {
...
+		unsafe {
+			let mut carry = 0u64;
+			let mut ret: [u64; $n_words] = mem::uninitialized();
+			let mut me_ptr = $self_expr.0.as_ptr();
+			let mut you_ptr = $other.0.as_ptr();
+			let mut ret_ptr = ret.as_mut_ptr();
+			for _ in 0..$n_words {
+				let (res1, overflow1) = (*me_ptr).overflowing_add(*you_ptr);
+
+				if carry != 0 {
```
Comment (NikVolf) [inline src/uint.rs:75]: i think we can avoid jump here

you can just add carry with overflow to res1 and write overflow flag back to carry, no need to check anything?

Comment (NikVolf) [reply src/uint.rs:75]: because if `overflow1 = true`, `res1 + carry` cannot overflow

Comment (NikVolf) [reply src/uint.rs:75]: so here what i had in mind?

https://github.com/debris/bigint/pull/1

it does not show any improvement on my windows thoe

Comment (debris) [reply src/uint.rs:75]: @NikVolf did you disable assembly for nightly? also, together with @Vurich we found a way to parallelize the operations, so the addition and substraction is almost 10x faster, and even faster than assembly code. For now, I'm closing the pr, and we will open the new one tomorrow

**Pull request closed without merging**